pySDC/helpers/fieldsIO.py

```diff
@@ -0,0 +1,731 @@
+    def setupMPI(cls, comm: MPI.Intracomm, iLocX, nLocX):
+        """
+        Setup the MPI mode for the files IO, considering a decomposition
+        of the 1D grid into contiuous subintervals.
+
+        Parameters
+        ----------
+        comm : MPI.Intracomm
+            The space decomposition communicator.
+        iLocX : int
+            Starting index of the local sub-domain in the global `gridX`.
+        nLocX : int
+            Number of points in the local sub-domain.
+        """
+        cls.comm = comm
+        cls.iLocX = iLocX
+        cls.nLocX = nLocX
+        cls.mpiFile = None
+
+    @property
+    def MPI_ON(self):
+        """Wether or not MPI is activated"""
+        if self.comm is None:
+            return False
+        return self.comm.Get_size() > 1
+
+    @property
+    def MPI_ROOT(self):
+        """Wether or not the process is MPI Root"""
+        if self.comm is None:
+            return True
+        return self.comm.Get_rank() == 0
+
+    def MPI_FILE_OPEN(self, mode):
+        """Open the binary file in MPI mode"""
+        amode = {
+            "r": MPI.MODE_RDONLY,
+            "a": MPI.MODE_WRONLY | MPI.MODE_APPEND,
+        }[mode]
+        self.mpiFile = MPI.File.Open(self.comm, self.fileName, amode)
+
+    def MPI_WRITE(self, data):
+        """Write data (np.ndarray) in the binary file in MPI mode, at the current file cursor position."""
+        self.mpiFile.Write(data)
+
+    def MPI_WRITE_AT(self, offset, data: np.ndarray):
+        """
+        Write data in the binary file in MPI mode, with a given offset
+        **relative to the beginning of the file**.
+
+        Parameters
+        ----------
+        offset : int
+            Offset to write at, relative to the beginning of the file, in bytes.
+        data : np.ndarray
+            Data to be written in the binary file.
+        """
+        self.mpiFile.Write_at(offset, data)
+
+    def MPI_READ_AT(self, offset, data):
+        """
+        Read data from the binary file in MPI mode, with a given offset
+        **relative to the beginning of the file**.
+
+        Parameters
+        ----------
+        offset : int
+            Offset to read at, relative to the beginning of the file, in bytes.
+        data : np.ndarray
+            Array on which to read the data from the binary file.
+        """
+        self.mpiFile.Read_at(offset, data)
+
+    def MPI_FILE_CLOSE(self):
+        """Close the binary file in MPI mode"""
+        self.mpiFile.Close()
+        self.mpiFile = None
+
+    def initialize(self):
+        """Initialize the binary file (write header) in MPI mode"""
+        if self.MPI_ROOT:
+            super().initialize()
+        if self.MPI_ON:
+            self.comm.Barrier()  # Important, should not be removed !
+            self.initialized = True
+
+    def addField(self, time, field):
+        """
+        Append one field solution at the end of the file with one given time,
+        eventually using MPI.
```

Comment:
You are using "eventually" incorrectly again 🙈

Reply:
At least there is one part of the code where I used it correctly